Ctauto 286

jrdavisllbean opened the pull request:
Hi @ecoronado89, @didierllbean and @Gamdie,

I have completed about 80% of CTTCM-2482 but unfortunately there are **a lot of changes**.

Would it be possible to code review my changes and provide feedback. After I commit the feedback then can we merge? Afterwards I will create a finish_ctauto_286 branch. 

Please see below for changes, I apologize there are so many. The good news is the checkout flow in this branch could be used for most tests (hopefully) so the commits should be smaller over time.

A. Checkout Flow
Added getOrdersummary() to ShippingAddress, BillingAddress, PaymentOptions, ReviewAndPlaceOrder and ThankYou pages. This will make verifying the data in steps 10-14 much easier.

B. AddedToYourShoppingBagModal Page
@didierllbean, I noticed when calling the getContinueBtn() and clickCheckoutBtn() methods that a TimeoutException was occuring after clicking Add to Bag from the Oxford Shirt Product Page. I modified these methods to handle a scenario where one button is visible, the other is hidden. Please see my comments in this class.

C. ShoppingBag
Added a beta get items total method. I will eventually add the CurrencyUtil code when I create.

D. ShoppingBagItem
Combined engraving/monogram methods because they have the same locators.

E. Added web GiftCards class for E-Gift Card and Gift Card.

F. Header
Added giftCard() method which returns the GiftCards object, getShoppingBagItemCount() which returns the number of items in the Shopping Bag link.

G. SearchResultPage
Added selectRandomProduct(int indexPosition), this method allows you to get a specific item in the results, e.g. selectRandomProduct(1);. I need to rename method to selectProduct(int indexPosition) in another commit.

To run the test, add com.llbean.automation.web.tests.CTTCM_2482 to the TestSuite.xml className tag. I have tested locally against Chrome and Firefox.

ecoronado89 commented on src/main/java/com/llbean/automation/web/pages/desktopNTablet/BillingAddress.java:
```diff
@@ -74,11 +75,19 @@
     @FindBy(id="avUseEntered")
```
Add PageLocators page to define web elements

ecoronado89 commented on src/main/java/com/llbean/automation/web/pages/desktopNTablet/GiftCards.java:
```diff
@@ -0,0 +1,149 @@
+
+    static final Logger logger = Logger.getLogger(GiftCards.class);
+
+    @FindBy(css = "li.activenavcategory > ul > li:nth-child(1) > h4 > a")
```
Web element definitions should be placed on PageLocators class

ecoronado89 commented on src/main/java/com/llbean/automation/web/tests/CTTCM_2482.java:
```diff
@@ -0,0 +1,167 @@
+
+        logger.info("Executing STEP 1 - \"Go to a product page eligible for *Monogramming*\"");
+        HomePage home = new HomePage(driver);
+        SearchResultPage searchResultPage = home.header.searchfor(TESTCASE_ITEM_NAME);
```
All search should be in searchData

ecoronado89 commented on src/main/java/com/llbean/automation/web/tests/CTTCM_2482.java:
```diff
@@ -0,0 +1,167 @@
+
+    final static Logger logger = Logger.getLogger(CTTCM_2482.class);
+
+    final static String CATALOG_QUICKSHOP = "Catalog Quickshop";
```
This type of variables should be in a properties file

jrdavisllbean commented:
I have added the latest feedback:
1) All text variables are in a properties file.
2) I created SearchData searchWinkleFreeOxfordShirts()
3) Added wait to SearchResultPage get product methods, this will resolve the race condition @didierllbean was seeing.
4) All page locators that were defined in the factory annotation have been moved to the PageLocators interface.

Note when reviewing my test I sometimes would get "org.openqa.selenium.WebDriverException: Element is not clickable at point......" exception when the ProductPage selectItem, e.g. selectItemSet(). Per dider, a wait needs to be added.
